### PR TITLE
added point on feature module

### PR DESCRIPTION
**Pull request by SteffenHaeussler**
merge request is part of: https://github.com/diogomatoschaves/pyturf/issues/14

Please make sure that:

- [x] You have read [CONTRIBUTING.md](../CONTRIBUTING.md)
- [x] You Include a link to the issue this pull request is associated to.
- [x] The linter has been run at the root level of the project: `black .`
- [x] All tests are passing: `python -m pytest --verbose --cov=./` 
- [x] Your branch is up to date with master: `git rebase upstream/master`

For new modules, also make sure that:

- [x] The new module function is exported in `turf/<your-module>/__init__.py` and `turf/__init__.py`.
- [x] The new module has been added under the `Available Modules` section on the [README](../README.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,8 @@ Currently, the following modules have been implemented:
 - [length](https://github.com/diogomatoschaves/pyturf/tree/master/turf/length)
 - [midpoint](https://github.com/diogomatoschaves/pyturf/tree/master/turf/midpoint)
 - [nearest-point](https://github.com/diogomatoschaves/pyturf/tree/master/turf/nearest_point)
+- [point-on-feature](https://github.com/diogomatoschaves/pyturf/tree/master/turf/point_on_feature)
+- [nearest-point](https://github.com/diogomatoschaves/pyturf/tree/master/turf/nearest_point)
```
Comment (diogomatoschaves) [inline README.md:87]: `nearest-point` is being repeated.

File: turf/__init__.py [hunk 2]
```diff
@@ -15,6 +15,8 @@
 from turf.length import length
 from turf.midpoint import midpoint
 from turf.nearest_point import nearest_point
+from turf.point_on_feature import point_on_feature
+from turf.nearest_point import nearest_point
```
Comment (diogomatoschaves) [inline turf/__init__.py:19]: `nearest-point` is being repeated.

File: turf/point_on_feature/_point_on_feature.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                for seg_index in range(1, len(line)):
+
+                    if point_on_segment(
+                        center_coords, line[seg_index - 1], line[seg_index]
+                    ):
+
+                        center_on_surface = True
+                        break
+
+        elif geometry_type == "Polygon" or geometry_type == "MultiPolygon":
```
Comment (diogomatoschaves) [inline turf/point_on_feature/_point_on_feature.py:73]: Could be `elif geometry_type in ["Polygon", "MultiPolygon"]:` for consistency

File: turf/point_on_feature/_point_on_feature.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+            if boolean_point_in_polygon(center_point, feature):
+                center_on_surface = True
+                break
+
+    if center_on_surface:
+        point_on_surface = center_point
+
+    else:
+        vertices = explode(feature_collection)
```
Comment (diogomatoschaves) [inline turf/point_on_feature/_point_on_feature.py:83]: From what I can tell the `nearest_point` module already uses the `explode` method internally, so this step is not needed.

File: turf/point_on_feature/_point_on_feature.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+        geojson = feature_collection([geojson])
+
+    else:
+        geojson = feature_collection([feature(geojson)])
+
+    return geojson
+
+
+def point_on_segment(point: List, segment_start: List, segment_end: List) -> bool:
```
Comment (diogomatoschaves) [inline turf/point_on_feature/_point_on_feature.py:111]: I see some overlap in this method and the, still not implemented, `boolean-point-on-line` module. Perhaps we can use that one here once it is.

File: turf/point_on_feature/_point_on_feature.py (hunk 4 above)
Comment (SteffenHaeussler) [reply turf/point_on_feature/_point_on_feature.py:83]: agree. the first thing in the nearest point method is to "explode" the feature. this is a duplicate. I remove it.

File: turf/__init__.py (hunk 2 above)
Comment (SteffenHaeussler) [reply turf/__init__.py:19]: fixed. this is an artefact of the rebase. My fault.

File: README.md (hunk 1 above)
Comment (SteffenHaeussler) [reply README.md:87]: fixed. this is an artefact of the rebase. My fault.

File: turf/point_on_feature/_point_on_feature.py (hunk 5 above)
Comment (SteffenHaeussler) [reply turf/point_on_feature/_point_on_feature.py:111]: I had a quick look in the source code and I agree. This makes sense to move this function into the `boolean-point-on-line` module. I created [issue 27](https://github.com/diogomatoschaves/pyturf/issues/27).

**Pull request merged**